Formula/gap.rb

```diff
@@ -0,0 +1,70 @@
+class Gap < Formula
+  desc "System for computational discrete algebra"
+  homepage "https://www.gap-system.org/"
+  url "https://www.gap-system.org/pub/gap/gap-4.9/tar.bz2/gap-4.9.1.tar.bz2"
+  version "4.9.1"
+  sha256 "d93abb263e6b9ce9f56bfb820ac263b9a0f349f6bd1584244c7b242931d0dd3a"
+
+  depends_on "gmp"
+  depends_on "readline"
+
+  def install
+    # Remove some unused files
+    rm Dir["bin/*.bat", "bin/*.ico", "bin/*.bmp", "bin/cygwin.ver"]
+
+    # XXX:  Currently there is no `install` target in `Makefile`.
+    #   According to the manual installation instructions in
+    #
+    #     https://github.com/gap-system/gap/blob/master/INSTALL.md
+    #
+    #   the compiled "bundle" is intended to be used "as is," and there is
+    #   no instructions for how to remove the source and other unnecessary
+    #   files after compilation.  Moreover, the content of the
+    #   subdirectories with special names, such as `bin` and `lib`, is not
+    #   suitable for merging with the content of the corresponding
+    #   subdirectories of `/usr/local`.  The easiest temporary solution seems
+    #   to be to drop the compiled bundle into `<prefix>/libexec` and to
+    #   create a symlink `<prefix>/bin/gap` to the startup script.
+    #   This use of `libexec` seems to contradict Linux Filesystem Hierarchy
+    #   Standard, but is recommended in Homebrew's "Formula Cookbook."
+
+    libexec.install Dir["*"]
+
+    # GAP does not support "make install" so it has to be compiled in place
+
+    cd libexec do
+      args = %W[--prefix=#{libexec} --with-gmp=system]
+      system "./configure", *args
+      system "make"
+    end
+    
+    # Create a symlink `bin/gap` from the `gap` binary
+    bin.install_symlink libexec/"gap" => "gap"
```

Comment:
Huh, is that really what we want? It's a bit unclear to me what is linked to what here, too...

Better might be to create `#{prefix}/bin/gap` as a shell script, similar (or basically identical) to what we have in `$srcdir/bin/gap.sh`, but of course with the correct paths inserted (this is what I plan to do for `make install`).

Reply:
This results in symlink `/usr/local/bin/gap -> ../Cellar/gap/4.10.0/bin/gap` where the latter is a symlink `/usr/local/Cellar/gap/4.10.0/bin/gap -> ../libexec/gap`. I thought we can do this since now `gap` can be run without options and it finds the GAP library location itself.